### DISCUSSION TITLE
Allow more POIs editing options (moving, changing category, ...)

### DISCUSSION THREAD

**Discussion by arnaudvergnet** [organicmaps/organicmaps#1249] [Ideas]
Organic maps editor is very handy and simple, but sometimes when seeing invalid information, I am forced to switch to other apps like Vespucci to make a simple edit. A restaurant tagged as fast food, a bus station on the wrong side of the road, and so on. This works but not very practical as editors like Vespucci are very hard to understand for casual users.

I am not asking for powerful features such as editing and creating new ways, this would not be a good idea. But rather some more options on POI editing. like moving a POI, changing its type, maybe entering an advanced mode to enter a custom tag.

Anyways, thanks a lot for this app, it is really a handy tool!

**Comment by biodranik** [comment]
Thanks! We would love to add these features, but there are many people who are not familiar with OSM and global map edits. They may easily break something or add something wrong, or even start spamming.

Any ideas how to prevent it are welcomed.

**Reply by arnaudvergnet** [reply]
Maybe try to teach the user by showing a short popup explaining the risks? [StreetComplete](https://github.com/streetcomplete/StreetComplete) does something like this when you try to edit a POI too far from your GPS position. The popup asks the user confirmation they are sure of the data they are going to enter.

If you want to keep new users from moving and changing too much stuff, you could do a system like StackOverflow where you earn functionalities the more you contribute. If your account does not have a lot of OSM edits, you can only edit POI. After a certain amount, you unlock changing categories then moving POIs. Maybe a last step for users with a lot of edits would be to enter tags manually, like [OsmAnd](https://github.com/osmandapp/OsmAnd) does.

As Organic maps is a lot easier to use than any other OSM app, this could be a good way to teach people how to use OSM, a sort of entry point into the project. You could even show little tips as the user edits data (which can be turned of because some could find it annoying).

Again, StreetComplete does something a bit like this, where they reward the user with external links to interesting projects the more you contribute. This is a small motivation to continue making edits.

**Reply by biodranik** [reply]
I like the idea of unlocking features for more "professional" users. There is only one problem I see: a spammer can quickly create hundreds of edits. Then these edits will be rolled back/fixed by the community. And there is no reliable way known for me to programmatically detect, that these edits were spam. Any ideas on how to distinguish "good" users from "bad" ones by analyzing their OSM history are welcomed!

**Reply by arnaudvergnet** [reply]
How does [Vespucci](https://github.com/MarcusWolschon/osmeditor4android) handle this? Because anyone can load the app and break everything.

**Reply by biodranik** [reply]
1) Its audience is mostly professional OSMers, while in MapsMe and OM it's mostly a "general user".
2) Its audience is very small compared to MapsMe, where I added the editor in 2015. And even compared to Organic Maps now.

**Reply by FinixFighter** [reply]
Well actually, if a person wants to break things, he/she can already do it.. you can create hundreds of POIs, you can rename POIs, and more.. this is sufficient for ruining an entire city in less than 24hours.. Anyway every OpenStreetMap contribuor has an attribute called "Active contributor" (take a look to my page: http://hdyc.neis-one.org/?FinixFighter) defined by two parameters (https://wiki.osmfoundation.org/wiki/Licence/Contributor_Terms#Rights_Granted); if both parameters are equal to "yes" then the user could be enabled to make some more advanced edit (moving POIs, changing category etc..). Obviously this idea could be integrated with something more :)

**Reply by TheAdventurer64** [reply]
Maybe a system for verifying edits. When I did Google Map editing, there was a feature so that another contributor needed to verify your edit before it got uploaded to the map.

**Reply by biodranik** [reply]
There is zero chance that someone will have resources to verify edits. The only reliable way is to help OSM community by filtering bad guys and by relying on their post-upload reviews and fixes.

**Reply by TheAdventurer64** [reply]
How does Wikipedia do it? We could possibly have user/IP blocks that could use OSM's database of active blocks.

**Reply by biodranik** [reply]
Who will maintain it? Who will spend their time on that? The best reviewers are other OSM users.

**Reply by Eiim** [reply]
In terms of controlling access, I'm a fan of overall mapping days, which can't be quickly spammed and are more directly indicative of experience. As for "active contributor", it seems silly to withhold access from, for example, [Steve](https://en.wikipedia.org/wiki/Steve_Coast), because he's only mapped [27 days](https://hdyc.neis-one.org/?Steve) in the past year.

**Reply by TheAdventurer64** [reply]
Any progress on this?

**Reply by biodranik** [reply]
See also #1556 and #1325

**Comment by hb0nd** [comment]
Perhaps integrating the [iD-Tagging-Schema](https://github.com/openstreetmap/id-tagging-schema) could be good, it would result in more presets for the user and would remove the need to maintain OM's presets

**Reply by csidirop** [reply]
I like that!

**Comment by gleb09** [comment]
I'm going to have to lean the other way. I think Organic Maps is really great. It runs and feels a lot better than other open maps. Focus on that and let other apps/services sort out map editing.

**Reply by TheAdventurer64** [reply]
The issue is that for iOS, there aren't any good POI editors for general use. There's OSMAnd and GoMap, but both can be too advanced for general mapping.

**Reply by j13m126** [reply]
There's also [Every Door](https://apps.apple.com/app/every-door/id1621945342) for iOS.

**Comment by csidirop** [comment]
In this case consider adding a switch in the settings, so that we have to actively activating that the editing option. That way new users don't get confused.

**Comment by TheAdventurer64** [comment]
Maybe we could draw inspiration from EveryDoor, which is both user friendly and powerful. 

Regarding the fact that most editors on OM break things, I think by making the editor more powerful, we could attract a different crowd: More seasoned mappers would be able to fix and edit things easier, which would bring more popularity and attention towards Organic Maps. Plus, (if I remember correctly), EveryDoor isn't available for F-Droid, making those who want a powerful but simple editor potentially more interesting in the software.

**Comment by rouelibre1** [comment]
It would be great to have more relevant field for POI types. I've been trying to add bicycle parking, but the field are mostly irrelevant for this POI type (name of the place, opening hours, website...).

I guess this is generic, but it would be great to get inspiration from apps like StreetComplete and Everydoor for example.

In my case, the lack of appropriate fields defeats the purpose of using the feature in the app.

**Reply by biodranik** [reply]
Let's fix invalid fields. They are manually assigned for each type at `data/editor.config`
